src/uct/tcp/tcp_iface.c

```diff
@@ -234,11 +238,22 @@ ucs_status_t uct_tcp_iface_set_sockopt(uct_tcp_iface_t *iface, int fd)
         return UCS_ERR_IO_ERROR;
     }
 
-    ret = setsockopt(fd, SOL_SOCKET, SO_SNDBUF, (void*)&iface->sockopt.sndbuf,
-                     sizeof(int));
-    if (ret < 0) {
-        ucs_error("Failed to set SO_SNDBUF on fd %d: %m", fd);
-        return UCS_ERR_IO_ERROR;
+    if (iface->sockopt.sndbuf) {
+        ret = setsockopt(fd, SOL_SOCKET, SO_SNDBUF, (void*)&iface->sockopt.sndbuf,
```

Comment:
maybe add setsockopt wrapper to ucs/sys/sock.{c,h}?

Reply:
done